### PR TITLE
MODFEE-253 Lost item processing fee can be saved with a value less than 0

### PR DESCRIPTION
**Pull request by bisecomsergiy**
### Purpose
Avoid entering values less then 0 to Lost item processing fee 
### Approach
Add validation
Add tests
### Learning
https://issues.folio.org/browse/MODFEE-253

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java [hunk 1]
```diff
@@ -94,7 +105,33 @@ public void putLostItemFeesPoliciesByLostItemFeePolicyId(String lostItemFeePolic
...
+
     PgUtil.put(TABLE_NAME, entity, lostItemFeePolicyId, okapiHeaders, vertxContext,
       PutLostItemFeesPoliciesByLostItemFeePolicyIdResponse.class, asyncResultHandler);
   }
+
+  private void handleError(LostItemFeePolicy entity,
+    Handler<AsyncResult<Response>> asyncResultHandler) {
+
+      ImmutablePair<String, String> errorInformation = new ImmutablePair<>("lostItemProcessingFee",
+        getAmountValueAsString(entity));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java:121]: Why do we need this pair? Can these values be inlined?

File: src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java [hunk 2]
```diff
@@ -94,7 +103,28 @@ public void putLostItemFeesPoliciesByLostItemFeePolicyId(String lostItemFeePolic
...
+        succeededFuture(respond422WithApplicationJson(createError(
+          NEGATIVE_VALUE_MESSAGE, "lostItemProcessingFee",
+          entity.getLostItemProcessingFee().getAmount().toString()))));
+  }
+
+  private boolean shouldRefuseWhenLostItemProcessingFeeNegative(LostItemFeePolicy entity) {
+    return Optional.ofNullable(entity)
+      .map(LostItemFeePolicy::getLostItemProcessingFee)
+      .map(MonetaryValue::isNegative)
+      .orElse(false);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java:128]: Make 1 method that returns boolean and handles the error

File: src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java (hunk 1 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java]: It is refactored, thanks!

File: src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java [hunk 3]
```diff
@@ -94,7 +105,33 @@ public void putLostItemFeesPoliciesByLostItemFeePolicyId(String lostItemFeePolic
...
+    if (containsNegativeValue(entity)) {
+      handleError(entity, asyncResultHandler);
+      return;
+    }
+
     PgUtil.put(TABLE_NAME, entity, lostItemFeePolicyId, okapiHeaders, vertxContext,
       PutLostItemFeesPoliciesByLostItemFeePolicyIdResponse.class, asyncResultHandler);
   }
+
+  private void handleError(LostItemFeePolicy entity,
```
[the remark this answers is not in the document]
Comment (bisecomsergiy) [reply src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java]: The method is removed

File: src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java (hunk 2 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/rest/impl/LostItemFeePoliciesAPI.java]: The methods are coupled, thanks.

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**